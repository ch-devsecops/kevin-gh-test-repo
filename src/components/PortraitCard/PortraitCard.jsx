import React, { useContext, useEffect, useRef, useState } from 'react';
import PropTypes from 'prop-types';
import {
  Box,
  Icon,
  Image,
  Markdown,
  MarkdownHeading,
  MediaItem,
  Optional,
  Tooltip,
  useOptionalVideo,
} from '@honda-canada/design-system-react';
import { useTranslation } from 'react-i18next';
import { ModelExplorationContext } from '../ModelExplorationContext';
import { getGtmTagValue } from '../../utils/gtmEvents';
import themeStyles from './PortraitCard.styles';

const Container = themeStyles.apply(Box, 'Container');
const MediaContainer = themeStyles.apply(Box, 'MediaContainer');
const HeadingContainer = themeStyles.apply(Box, 'HeadingContainer');
const TextContainer = themeStyles.apply(Box, 'TextContainer');
const BodyTextContainer = themeStyles.apply(Box, 'BodyTextContainer');
const CTAGroup = themeStyles.apply(Box, 'CTAGroup');

const PortraitCard = ({
  title,
  bodyText,
  ctas,
  image,
  textColor = 'typographyDefault',
  contentAlignment = 'center',
  backgroundColor,
  iconImage,
  video = {},
  isTertiaryCTAGroup = false,
  textContainerMargins,
  gtmTags = {},
  toolTip,
}) => {
  const { optionalVideo } = useOptionalVideo(video);
  const tooltipContainer = useRef(null);
  const [maxWidth, setMaxWidth] = useState('none');

  const modelExplorationContext = useContext(ModelExplorationContext) || {};
  const { isDark } = modelExplorationContext;
  const { t } = useTranslation();

  useEffect(() => {
    if (typeof window === 'undefined') return;
    const handleResize = () => {
      const tooltipContainerWidth = tooltipContainer?.current?.offsetWidth || 0;

      if (tooltipContainerWidth > 0 && tooltipContainerWidth < 315) {
        setMaxWidth('180px');
      }
    };

    window.addEventListener('resize', handleResize);

    handleResize();
    // Remove event listener on cleanup
    return () => window.removeEventListener('resize', handleResize);
  }, []);

  const renderCTAs = isTertiary => {
    if (!ctas) {
      return null;
    }

    return (
      <CTAGroup
        isTertiaryCTAGroup={isTertiary}
        contentAlignment={contentAlignment}
        color={isDark ? 'white' : textColor}
      >
        {ctas}
      </CTAGroup>
    );
  };
  const ctaGroup = renderCTAs(isTertiaryCTAGroup);

  return (
    <Container
      backgroundColor={isDark ? '#000000' : backgroundColor}
      textAlign={contentAlignment}
      data-gtm-component-type={getGtmTagValue(gtmTags?.type)}
      data-gtm-category={getGtmTagValue(gtmTags?.category)}
      ref={tooltipContainer}
    >
      <Optional when={image}>
        <MediaContainer>
          <MediaItem
            image={<Image src={image?.props?.src} alt={image?.props?.alt} />}
            cmsIcon={iconImage}
            video={optionalVideo}
          />
        </MediaContainer>
      </Optional>
      <Optional when={!image}>
        <Box
          pt={['6px', '19px']}
          px={textContainerMargins && ['20px', '32px']}
          justifyContent={contentAlignment}
          display="flex"
        >
          <MediaItem image={iconImage} />
        </Box>
      </Optional>
      <Optional when={title || bodyText || (ctas && ctas.length > 0)}>
        <TextContainer hasMargins={textContainerMargins} hasImage={!!image}>
          <Optional when={title}>
            <HeadingContainer isDark={isDark} display="flex" justifyContent={contentAlignment}>
              <MarkdownHeading headingOverride="h4" color={isDark ? 'white' : textColor}>
                {title}
              </MarkdownHeading>
              <Optional when={toolTip}>
                <Tooltip
                  maxWidth={maxWidth}
                  styling={backgroundColor === 'white' || !backgroundColor ? 'light' : 'dark'}
                  content={
                    <Markdown size="xSmall" color="inherit">
                      {toolTip}
                    </Markdown>
                  }
                  ml="xxs"
                  forceHorizontalPosition="left"
                  ariaLabel={t('Shared.Common.showTooltipPopupAria')}
                  closeAriaLabel={t('Shared.Common.pressToClosePopupAria')}
                >
                  {({ active }) => (
                    <Box
                      width="16px" // iOS mobile / tablet fix
                      height="16px"
                    >
                      <Icon
                        name="information"
                        filled={active}
                        inverted={backgroundColor && backgroundColor !== 'white'}
                      />
                    </Box>
                  )}
                </Tooltip>
              </Optional>
            </HeadingContainer>
          </Optional>
          <Optional when={bodyText}>
            <BodyTextContainer isDark={isDark}>
              <Markdown color={isDark ? 'white' : textColor}>{bodyText}</Markdown>
            </BodyTextContainer>
          </Optional>
          {ctaGroup}
        </TextContainer>
      </Optional>
    </Container>
  );
};

PortraitCard.propTypes = {
  video: PropTypes.shape({
    src: PropTypes.string,
    ariaLabel: PropTypes.string,
    closeAriaLabel: PropTypes.string,
    onPlay: PropTypes.func,
  }),
  /**
   * Heading text (Markdown enabled)
   */
  title: PropTypes.string,
  image: PropTypes.element,
  /**
   * Content text (Markdown enabled)
   */
  bodyText: PropTypes.string,
  toolTip: PropTypes.string,
  /**
   * Sets text container background color.
   */
  backgroundColor: PropTypes.oneOf(['black', 'white', 'honda black', 'grey.5']),
  // Sets text color
  textColor: PropTypes.oneOf(['black', 'white']),
  /**
   * Content driven Icon
   */
  iconImage: PropTypes.element,
  contentAlignment: PropTypes.oneOf(['left', 'center']),
  ctas: PropTypes.arrayOf(PropTypes.element),
  isTertiaryCTAGroup: PropTypes.bool,
  textContainerMargins: PropTypes.bool,
  gtmTags: PropTypes.shape({
    type: PropTypes.string,
    category: PropTypes.string,
  }),
};

export default PortraitCard;
