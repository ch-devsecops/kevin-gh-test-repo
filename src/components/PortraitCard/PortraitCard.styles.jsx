import css from '@styled-system/css';
import { ThemeStyles } from '@honda-canada/design-system-react';
import { MC_THEME_NAME } from '../../utils/constants';

const themeStyles = new ThemeStyles();

themeStyles.addBaseStyles('Container')({
  display: 'flex',
  flexDirection: 'column',
  height: '100%',
});

themeStyles.addBaseStyles('MediaContainer')(
  css({
    width: '100%',
    '> div > img': {
      maxHeight: ['initial', '345px'],
    },
  }),
);

themeStyles.addBaseStyles('TextContainer')(
  {
    display: 'flex',
    flexDirection: 'column',
    flexGrow: '1',
  },
  ({ hasMargins, hasImage }) => {
    const styles = {
      py: hasImage ? ['20px', 'l'] : ['6px', '13px'],
    };

    if (hasMargins) {
      styles.px = ['20px', 'l'];
    } else {
      styles.paddingRight = ['20px', '40px'];
      styles.paddingLeft = ['20px', 0];
    }

    return css(styles);
  },
);

themeStyles.addBaseStyles('HeadingContainer')(({ isDark }) =>
  css({
    mb: 'm',
    '& a': {
      color: isDark ? 'white' : undefined,
    },
  }),
);

themeStyles.addBaseStyles('CTAGroup')(({ isTertiaryCTAGroup, contentAlignment }) => {
  const ctaStyles = {
    mt: 'm',
    mr: 'xs',
    textAlign: contentAlignment,
  };

  if (contentAlignment === 'center') {
    ctaStyles.ml = 'xs';
  }

  if (isTertiaryCTAGroup) {
    ctaStyles.display = 'inline-block';
  }

  return css({
    display: ['flex', 'initial'],
    justifyContent: contentAlignment,
    alignItems: 'center',
    flexWrap: 'wrap',
    flexDirection: [isTertiaryCTAGroup ? 'row' : 'column', 'row'],

    '& > *': ctaStyles,
  });
});

themeStyles.addBaseStyles('BodyTextContainer')(({ isDark }) =>
  css({
    mb: ['xs', 'm'],
    flexGrow: 1,

    '&:last-child': {
      mb: '0px',
    },
    h6: {
      color: 'blue',
    },
    '& a': {
      color: isDark ? 'white' : undefined,
    },
  }),
);

themeStyles.addThemeStyles(
  'BodyTextContainer',
  MC_THEME_NAME,
)(
  css({
    h6: {
      color: ['black'],
    },
  }),
);
export default themeStyles;
