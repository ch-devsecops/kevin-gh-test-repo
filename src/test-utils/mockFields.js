/* eslint-disable max-len */
/* eslint-disable max-classes-per-file */

// FIXME: jsdoc popup not working on test files
/** Class representing a mock data. */
class MockData {
  /**
   * Add a field.
   * @param {string} key - key of the field obj: i.e. 'title'
   * @param {Object} value - value of the field obj: i.e. { value: 'some text' }.
   */
  add(key, value) {
    this[key] = value;
  }
}

/**
 * Class representing mock fields object.
 * @extends MockData
 *
 * available method: add()
 *
 * default fields: [anchorId, title, bodyText, image, ctaLink, ctaType, ctaIcon, gtmCategory, gtmTitle, contentTitle, subtitle]
 *
 * note: we can add more commonly used fields in the class declaration if needed
 */
class MockFields extends MockData {
  constructor() {
    super();

    this.title = { value: 'Title' };
    this.bodyText = { value: 'BodyText' };
    this.image = {
      value: { src: 'mock.png', alt: 'mock alt' },
    };
    this.ctaLink = {
      value: {
        href: 'www.source.ca',
        title: 'cta aria label',
        text: 'CTA Text',
      },
    };
    this.ctaType = {
      value: 'Tertiary',
    };
    this.ctaIcon = {
      fields: {
        value: {
          value: 'arrowRight',
        },
      },
    };
    this.gtmCategory = {
      fields: {
        value: {
          value: 'Others',
        },
      },
    };
    this.gtmTitle = {
      value: 'gtm title',
    };
    this.contentTitle = {
      value: 'Content Title',
    };
    this.subtitle = {
      value: 'Content Subtitle',
    };
    this.seoH1 = {
      value: '',
    };
    this.anchorId = {
      value: 'anchor-id',
    };
    this.ctaLabel = {
      value: 'CTA Label',
    };
  }
}

/**
 * Class representing a mock rendering object
 * @extends MockData
 *
 * available method: add()
 */
class MockRendering extends MockData {
  /**
   * Create a mock rendering object
   * @param {string} componentName - adds componentName
   */
  constructor(componentName) {
    super();

    this.componentName = componentName;
  }
}

export { MockFields, MockRendering };
