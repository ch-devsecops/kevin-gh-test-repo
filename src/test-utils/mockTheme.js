/**
 * Mock styled-components theme, if you want to test how your component
 * implements theme styles. Import as-is, or spread in any
 * additional values you need.
 *
 * Example usage:

  it('should apply the bgColor param', () => {
    const { colors } = MockTheme;
    const { container } = render(
      <MyComponent
        fields={mockFields}
        params={{ bgColor: 'Gray' }}
      />,
    );
    const wrapper = container.querySelector('section');

    expect(wrapper).toHaveStyleRule('background-color', colors.grey[5]);
  });
 */

import { HONDA_THEME_NAME } from '../utils/constants';

const mockTheme = {
  letterSpacings: {
    sm: '0.5px',
    md: '1px',
  },
  lineHeights: {
    sm: '22px',
    md: '26px',
  },
  colors: {
    white: '#fff',
    transparent: 'transparent',
    grey: ['#707070', '#C7C7C7', '#D5D5D5', '#EAEAEA', '#F8F8F8', '#F5F5F5'],
    black: '#2C2C2C',
    red: '#DA2C2E',
    redShades: ['rgba(200, 16, 46, 0.1)'],
    aodaRed: '#D92629',
    lightRed: '#EC9597',
    darkRed: '#A21C1D',
    blue: '#334C96',
    darkBlue: '#002D62',
    lightBlue: '#00447826',
    yellow: ['#F9CA52', '#EDAB00', '#8D763F'],
    fadedBlue: '#ADB7D5',
    primary: '#DA2C2E',
    primaryAoda: '#D92629',
    primaryDark: '#A21C1D',
    primaryLight: '#EC9597',
    secondary: '#2C2C2C',
    secondaryLight: '#C7C7C7',
    tertiary: '#002D62',
    tertiaryLight: '#334C96',
    tertiaryDisabled: '#ADB7D5',
    danger: '#DA2C2E',
    warning: '#2C2C2C',
    warningLight: '#EAEAEA',
    success: '#334C96',
    successLight: '#d9e0e7',
    disabled: '#C7C7C7',
    typographyDefault: '#2C2C2C',
  },
  space: [0, 4, 8, 12, 16, 24, 32, 48, 64, 96, 128, 144, 192, 256],
  borders: [0, '1px solid', '2px solid', '3px solid'],
  breakpoints: ['768px', '992px', '1200px'],
  fontWeights: {
    regular: 'normal',
    bold: 700,
  },
  fontSizes: {
    sm: '12px',
  },
  shadows: {
    smooth: {
      md: '0.1px 0.3px 0.3px rgba(0, 0, 0, 0.02),0.3px 0.7px 0.9px rgba(0, 0, 0, 0.028),0.7px 1.4px 1.8px rgba(0, 0, 0, 0.035),1.5px 2.9px 3.7px rgba(0, 0, 0, 0.047),4px 8px 10px rgba(0, 0, 0, 0.07)',
    },
    box: {
      subNav: '0px 2px 4px rgba(0, 0, 0, 0.1)',
    },
  },
  zIndices: {
    accordionItemTitle: 200,
    modal: 13001,
    sectionNav: 13000,
  },
  transitionDuration: {
    t1: '1s',
  },
  transitionTimingFunction: {
    default: 'ease-in-out',
    out: 'ease-out',
    in: 'ease-in',
  },
  name: HONDA_THEME_NAME,
  fonts: {
    default: 'Helvetica',
    bold: 'Cooper Black',
    heading: 'Roboto',
  },
  header: {
    desktop: {},
    mobile: {},
  },
  icons: {},
  logIconWarnings: false,
};

export default mockTheme;
