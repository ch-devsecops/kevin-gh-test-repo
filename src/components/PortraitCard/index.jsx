import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import PropTypes from 'prop-types';
import { Fade } from '@honda-canada/design-system-react';
import { InView } from 'react-intersection-observer';
import {
  getVideoProps,
  colourTokenForParam,
  mapGTMCategory,
  foregroundColourTokenForParam,
  mapFieldToDesignSystemImage,
  mapJssFieldsToCtaComponents,
} from '../../utils/sitecoreFields';
import { ModelExplorationContext } from '../ModelExplorationContext';
import PortraitCard from './PortraitCard';
import { JSSFieldPropType } from '../../utils/propTypes';
import { useAppName } from '../../utils/sitecoreContext';
import { PSP_SITE_NAME } from '../../utils/constants';

const PortraitCardWrapper = ({ fields = {}, params = {}, rendering }) => {
  const { t } = useTranslation();
  const appName = useAppName();

  const { mediaIcon, mediaImage, videoUrl, bodyText, title, gtmCategory, gtmTitle, toolTip } = fields;

  const { contentBgColour, contentAlignment, hasPadding } = params;

  // TODO: confirm aria-label approach
  const video =
    videoUrl &&
    getVideoProps(videoUrl?.value, t('Shared.Common.playVideoAria'), t('Shared.Common.closeVideoModalAria'));
  const modelExplorationContext = useContext(ModelExplorationContext) || {};
  const { isDark } = modelExplorationContext;
  const image = mapFieldToDesignSystemImage(mediaImage);
  const cmsIcon = mapFieldToDesignSystemImage(mediaIcon);

  switch (appName) {
    case PSP_SITE_NAME:
      fields.gtmInteractionType1 = { value: 'cta: explore' };
      fields.gtmInteractionType2 = { value: 'cta: explore' };
      fields.gtmInteractionType3 = { value: 'cta: explore' };
      break;
    default:
      break;
  }

  const ctas = mapJssFieldsToCtaComponents(fields, gtmTitle?.value, rendering?.componentName);
  const gtmTags = {
    type: rendering?.componentName,
    category: mapGTMCategory(gtmCategory),
    title: title?.value,
  };

  return (
    <InView triggerOnce>
      {({ inView, ref }) => (
        <Fade ref={ref} shouldAnimate={inView} height="100%" initialOpacity={0}>
          <PortraitCard
            title={title?.value}
            bodyText={bodyText?.value}
            backgroundColor={colourTokenForParam[contentBgColour?.toLowerCase()]}
            contentAlignment={contentAlignment?.toLowerCase()}
            isTertiaryCTAGroup={fields?.ctaType1?.value?.toLowerCase() === 'tertiary'}
            textColor={isDark ? 'white' : foregroundColourTokenForParam[contentBgColour?.toLowerCase()]}
            image={image}
            ctas={ctas}
            video={video}
            iconImage={cmsIcon}
            textContainerMargins={hasPadding && hasPadding === 1}
            gtmTags={gtmTags}
            toolTip={toolTip?.value}
          />
        </Fade>
      )}
    </InView>
  );
};

PortraitCardWrapper.defaultProps = {
  fields: {},
  params: {},
};

PortraitCardWrapper.propTypes = {
  fields: PropTypes.shape({
    gtmCategory: PropTypes.shape({
      fields: PropTypes.shape({
        value: JSSFieldPropType,
      }),
    }),
    gtmTitle: JSSFieldPropType,
    mediaIcon: PropTypes.shape({
      value: PropTypes.shape({
        src: PropTypes.string,
        alt: PropTypes.string,
        width: PropTypes.string,
        height: PropTypes.string,
      }),
    }),
    mediaImage: PropTypes.shape({
      value: PropTypes.shape({
        src: PropTypes.string,
        alt: PropTypes.string,
        width: PropTypes.string,
        height: PropTypes.string,
      }),
    }),
    videoUrl: JSSFieldPropType,
    bodyText: JSSFieldPropType,
    title: JSSFieldPropType,
    toolTip: JSSFieldPropType,
    ctaType1: JSSFieldPropType,
  }),
  params: PropTypes.shape({
    contentBgColour: PropTypes.string,
    contentAlignment: PropTypes.string,
    hasPadding: PropTypes.string,
  }),
  rendering: PropTypes.shape({
    componentName: PropTypes.string,
  }),
};

export default PortraitCardWrapper;
