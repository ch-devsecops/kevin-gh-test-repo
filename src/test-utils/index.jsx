import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import { BrowserRouter } from 'react-router-dom';
import { OptionalVideoModalContextProvider } from '@honda-canada/design-system-react';
import UserLocationProvider from '@honda-canada/user-location';
import { configureAxe } from 'jest-axe';
import ModelExplorationContextProvider from '../components/ModelExplorationContext';
import ModelFiltersContextProvider from '../components/ModelFiltersContext';
import { MockFields, MockRendering } from './mockFields';
import MockTheme from './mockTheme';
import 'intersection-observer';
import { ACURA_THEME_NAME } from '../utils/constants';

const generateJssWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ACURA_THEME_NAME },
          ...context?.sitecoreContext,
        }}
        componentFactory={{}}
      >
        <BrowserRouter>
          <UserLocationProvider>
            <ThemeProvider theme={context?.theme || MockTheme}>
              <OptionalVideoModalContextProvider>
                <ModelExplorationContextProvider>
                  <ModelFiltersContextProvider>{children}</ModelFiltersContextProvider>
                </ModelExplorationContextProvider>
              </OptionalVideoModalContextProvider>
            </ThemeProvider>
          </UserLocationProvider>
        </BrowserRouter>
      </SitecoreContext>
    );
  };
/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in an typical JSS app's context providers.
 * Note that you can include a `context` object to set or override values
 * for SitecoreContext and ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.context] - default values for SitecoreContext, ThemeProvider
 * @param {string} [options.context.sitecoreContext] - value for SitecoreContext
 * @param {string} [options.context.theme] - value for ThemeProvider
 * @example
 * const { getByText } = jssRender(
 *  <MyComponent />, { context: { sitecoreContext: { language: 'fr' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const jssRender = (component, options = {}) => {
  const { context } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.context;

  return render(component, {
    wrapper: generateJssWrapper(context),
    ...options,
  });
};

const generateWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <BrowserRouter>
        <ThemeProvider theme={context?.theme || MockTheme}>
          <OptionalVideoModalContextProvider>{children}</OptionalVideoModalContextProvider>
        </ThemeProvider>
      </BrowserRouter>
    );
  };

/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in typical design-system context providers.
 * Note that you can include a `theme` object to set or override values
 * for ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.theme] - default value for ThemeProvider
 * @example
 * const { getByText } = render(
 *  <MyComponent />, { theme: { space: { wrapperGap: '15px' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const customRender = (component, options = {}) => {
  const { theme } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.theme;

  return render(component, {
    wrapper: generateWrapper(theme),
    ...options,
  });
};

// disable accessibility checks for errors that are usually trigged by content mistakes
const axe = configureAxe({
  rules: {
    'duplicate-id': { enabled: false },
    'heading-order': { enabled: false },
  },
});

export * from 'jest-axe';
export { axe };
export * from '@testing-library/react';
export { customRender as render, jssRender };
export { MockFields, MockRendering, MockTheme };
