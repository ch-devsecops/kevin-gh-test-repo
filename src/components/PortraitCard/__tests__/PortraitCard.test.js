import React from 'react';
import { axe, cleanup, render } from '../../../test-utils';
import PortraitCard from '../PortraitCard';

afterEach(cleanup);

describe('<PortraitCard>', () => {
  it('should render correctly with basic props', async () => {
    const { container } = render(
      <PortraitCard
        title="The title"
        bodyText="My body text"
        // eslint-disable-next-line react/jsx-key
        ctas={[<a href="https://example.com">CTA</a>]}
        image={<img src="https://via.placeholder.com/294x195" alt="mock alt" />}
        toolTip="The tool tip"
      />,
    );

    expect(await axe(container)).toHaveNoViolations();
    expect(container).toMatchSnapshot();
  });
});
