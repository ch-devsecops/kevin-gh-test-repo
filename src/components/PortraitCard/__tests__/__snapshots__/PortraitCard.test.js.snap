// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PortraitCard> should render correctly with basic props 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 sc-1k44dfv-0 hlPDKA"
  >
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 eNCdgo"
    >
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 jbBrzI"
      >
        <img
          alt="mock alt"
          class="sc-1k44dfv-0 jLTFZk"
          draggable="true"
          loading="lazy"
          src="https://via.placeholder.com/294x195"
        />
      </div>
    </div>
    <div
      class="Box-sc-1bsf784-0 sc-1k44dfv-0 iNhCNC"
    >
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 dcHvEP"
        display="flex"
      >
        <div
          class="sc-1k44dfv-0 goPGQt"
          color="typographyDefault"
        >
          

          <h4
            class="sc-1k44dfv-0 fINFhE"
          >
            The title
          </h4>
        </div>
        <div
          class="Box-sc-1bsf784-0 jIegoo"
          height="14px"
          width="14px"
        >
          <button
            aria-label="Shared.Common.showTooltipPopupAria"
            class="sc-1k44dfv-0 jOKBEK"
          >
            <div
              class="Box-sc-1bsf784-0 dVQVTJ"
              height="16px"
              width="16px"
            />
          </button>
        </div>
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 vwhQc"
      >
        <div
          class="sc-1k44dfv-0 goPGQt"
          color="typographyDefault"
        >
          

          <p>
            My body text
          </p>
        </div>
      </div>
      <div
        class="Box-sc-1bsf784-0 sc-1k44dfv-0 gBifkq"
        color="typographyDefault"
      >
        <a
          href="https://example.com"
        >
          CTA
        </a>
      </div>
    </div>
  </div>
</div>
`;
